crates/stages/src/stages/headers.rs

```diff
@@ -279,10 +279,7 @@ mod tests {
         let rx = runner.execute(input);
         runner.consensus.update_tip(H256::from_low_u64_be(1));
         let result = rx.await.unwrap();
-        assert_matches!(
-            result,
-            Ok(ExecOutput { done: false, reached_tip: false, stage_progress: 100 })
-        );
+        assert_matches!(result, Err(StageError::Recoverable(_)));
```

Comment:
TODO: Let's further match these by downcasting but fine for now.

```suggestion
        // TODO: Downcast the internal error and actually check it
        assert_matches!(result, Err(StageError::Recoverable(_)));
```